packages/api/models/collections.js

```diff
@@ -1,253 +0,0 @@
-const { validateCollection } = require('../lib/utils');
-
-// Fields which are no longer supported in collection items, and which should
-// not be returned when records are read from the database.
-const deprecatedFields = Object.freeze([
-  'provider_path',
-]);
-
-class Collection {
-  /**
-   * Creates a new Collection model for managing storage and retrieval of
-   * collections against a DynamoDB table. The name of the table is specified
-   * by the environment variable `CollectionsTable`.  The table is partitioned
-   * by collection `name`, with `version` as the sort key, both of which are of
-   * type `S`.  The table schema is defined by the
-   * {@link collectionSchema collection schema}.
-   *
-   * Collections created by this model are also put into a
-   * {@link CollectionConfigStore} upon {@link #create creation} and removed
-   * from it when {@link #delete deleted}.  The store is
-   * {@link CollectionConfigStore#constructor created} by using the S3 bucket
-   * name and CloudFormation stack name given by the values of the environment
-   * variables `system_bucket` and `stackName`, respectively.
-   *
-   * @see Manager#constructor
-   */
-  constructor() {
-    this.dynamoDbClient = new Manager({
-      tableName: process.env.CollectionsTable,
-      tableHash: { name: 'name', type: 'S' },
-      tableRange: { name: 'version', type: 'S' },
-      schema: collectionSchema,
-    });
-
-    this.collectionConfigStore = new CollectionConfigStore(
-      process.env.system_bucket,
-      process.env.stackName
-    );
-  }
-
-  createTable() {
-    return this.dynamoDbClient.createTable();
-  }
-
-  deleteTable() {
-    return this.dynamoDbClient.deleteTable();
-  }
-
-  async get({ name, version }) {
-    const fetchedCollection = await this.dynamoDbClient.get({ name, version });
-
-    return omit(fetchedCollection, deprecatedFields);
-  }
-
-  /**
-   * Returns `true` if the collection with the specified name and version
-   * exists; `false` otherwise.
-   *
-   * @param {string} name - collection name
-   * @param {string} version - collection version
-   * @returns {boolean} `true` if the collection with the specified name and
-   *    version exists; `false` otherwise
-   */
-  async exists(name, version) {
-    return await this.dynamoDbClient.exists({ name, version });
-  }
-
-  /**
-   * Creates the specified collection and puts it into the collection
-   * configuration store that was specified during this model's construction.
-   * Uses the specified item's `name` and `version` as the key for putting the
-   * item in the config store.
-   *
-   * @param {Object} item - the collection configuration
-   * @param {string} item.name - the collection name
-   * @param {string} item.version - the collection version
-   * @returns {Promise<Object>} the created record
-   * @see #constructor
-   * @see Manager#create
-   * @see CollectionConfigStore#put
-   */
-  async createItem(item) {
-    validateCollection(item);
```

Comment:
Did we preserve API validation behaviors on create for this model?  I  ran into this during the granule model deletion, this is the only location the api schema validation occurs on write.

Reply:
We have preserved the schema validation in the POST endpoint in the previous ticket.